src/providers/ipfs/dummy.py

```diff
@@ -0,0 +1,33 @@
+import hashlib
+
+from web3.module import Module
+
+from .types import CIDv0, CIDv1, IPFSProvider, NotFound
+
+
+class DummyIPFSProvider(IPFSProvider, Module):
+    """Dummy IPFS provider which using the local filesystem as a backend"""
+
+    mempool: dict[CIDv0 | CIDv1, bytes]
```

Comment:
`cache`? mislead a little at first glance

Reply:
It's in-memory pool of content for this provider, that's why the name.